src/components/Table/sortRows.ts

```diff
@@ -27,21 +27,47 @@ function sortBatch<R extends Kinded>(
   caseSensitive: boolean,
 ): GridDataRow<R>[] {
   // When client-side sort, the sort value is the column index
-  const [value, direction] = sortState;
+  const [value, direction, persistent] = sortState;
+  
   const column = columns[value];
   const invert = direction === "DESC";
 
+  let persistentColumn: GridColumn<R, {}>;
+  if (persistent) {
+    persistentColumn = columns[persistent]
+  }
+
   // Make a shallow copy for sorting to avoid mutating the original list
   return [...batch].sort((a, b) => {
     const v1 = sortValue(applyRowFn(column, a, {} as any, 0), caseSensitive);
     const v2 = sortValue(applyRowFn(column, b, {} as any, 0), caseSensitive);
     const v1e = v1 === null || v1 === undefined;
     const v2e = v2 === null || v2 === undefined;
+
+    let p1 = false;
+    let p2 = false;
+    if (persistentColumn) {
+      p1 = sortValue(applyRowFn(persistentColumn, a, {} as any, 0), caseSensitive);
+      p2 = sortValue(applyRowFn(persistentColumn, b, {} as any, 0), caseSensitive);
+    }
     if ((a.pin || b.pin) && !(a.pin === b.pin)) {
       const ap = a.pin === "first" ? -1 : a.pin === "last" ? 1 : 0;
       const bp = b.pin === "first" ? -1 : b.pin === "last" ? 1 : 0;
       return ap === bp ? 0 : ap < bp ? -1 : 1;
-    } else if ((v1e && v2e) || v1 === v2) {
+    } else if (p1 && p2){
+      if ((v1e && v2e) || v1 === v2) {
```

Comment:
my thought :)

use favoriting for context:
we have several items favorited. then we sort a list based on a different value, we would still like to see those items favorited at the top however sorted based on the sort we selected 

here sorted based off name, with favorites at the top

<img width="1143" alt="Screen Shot 2022-07-12 at 2 44 54 PM" src="https://user-images.githubusercontent.com/100221471/178591990-3f75c27b-565e-4a16-a7ba-ec750134485a.png">

Reply:
I see why this works now; we're assuming p1/p2 are boolean values, i.e. this only handles favorites.

I don't really like that assumption, as it's just creating work/technical debt for the next person who is going to want to persistent sort by a non-boolean column, but I won't block the PR if that's what we decide to do.

Reply:
Fwiw maybe extra a `compare` function and then invoke it twice:

```
function sortBatch<R extends Kinded>(
  columns: GridColumn<R>[],
  batch: GridDataRow<R>[],
  sortState: SortState<number>,
  caseSensitive: boolean,
): GridDataRow<R>[] {
  // When client-side sort, the sort value is the column index
  const [value, direction] = sortState;
  const column = columns[value];
  const invert = direction === "DESC";

  // Make a shallow copy for sorting to avoid mutating the original list
  return [...batch].sort((a, b) => {
    if ((a.pin || b.pin) && !(a.pin === b.pin)) {
      const ap = a.pin === "first" ? -1 : a.pin === "last" ? 1 : 0;
      const bp = b.pin === "first" ? -1 : b.pin === "last" ? 1 : 0;
      return ap === bp ? 0 : ap < bp ? -1 : 1;
    } else if (persistentColumn) {
      const pv = compare(persistentColumn, a, b, false, caseSensitive);
      const v = compare(column, a, b, invert, caseSensitive);
      return pv ?? v;
    } else {
      return compare(persistentColumn, a, b, invert, caseSensitive);
    }
    return 0;
  });
}

function compare(column: any, a: GridDataRow<any>, b: GridDataRow<any>, invert: boolean, caseSensitive: boolean) {
  const v1 = sortValue(applyRowFn(column, a, {} as any, 0), caseSensitive);
  const v2 = sortValue(applyRowFn(column, b, {} as any, 0), caseSensitive);
  const v1e = v1 === null || v1 === undefined;
  const v2e = v2 === null || v2 === undefined;
  if ((v1e && v2e) || v1 === v2) {
    return 0;
  } else if (v1e || v1 < v2) {
    return invert ? 1 : -1;
  } else if (v2e || v1 > v2) {
    return invert ? -1 : 1;
  } else {
    return 0;
  }
}
```